src/redux/util.js

```diff
@@ -38,40 +38,75 @@ export function sortByString(list, key) {
   return sortByKey(list, elem => elem[key].toLowerCase());
 }
 
-export function sortByKey(list, key, order='asc') {
+export function sortByKey(list, key, order = 'asc') {
   return _.orderBy(list, [key], [order]);
 }
 
-export async function fetchAllRecordsUsingFetchWithBQ(fetchWithBQ, url, itemsPerPage, otherParams = null) {
-  const fetch = async (page) => {
+export async function fetchAllRecordsUsingFetchWithBQ(
+  fetchWithBQ,
+  url,
+  itemsPerPage,
+  otherParams = null,
+) {
+  const fetch = async page => {
     const params = new URLSearchParams({
       pagination: true,
       page,
       itemsPerPage,
       ...otherParams,
     });
-    const result = await fetchWithBQ(`${url}?${params.toString()}`);
-    return result.data;
 
+    try {
+      const result = await fetchWithBQ(`${url}?${params.toString()}`);
+      return { data: result.data, error: null };
+    } catch (err) {
+      return { data: null, error: err };
+    }
   };
-  const firstRs = await fetch(1);
 
-  if (!Object.hasOwn(firstRs, 'hydra:totalItems') || firstRs['hydra:totalItems'] <= firstRs['hydra:member'].length) {
+  const firstPageResult = await fetch(1);
+  if (firstPageResult.error) {
+    return { error: firstPageResult.error };
+  }
+
+  const firstPageData = firstPageResult.data;
+  if (
+    !Object.hasOwn(firstPageData, 'hydra:totalItems') ||
+    firstPageData['hydra:totalItems'] <= firstPageData['hydra:member'].length
+  ) {
     // Total items were already returned in the 1st request!
-    return firstRs['hydra:member'];
+    return {
+      data: firstPageData['hydra:member'],
+    };
   }
 
   // OK more pages are needed to be fetched to get all items..!
-  const totalItems = firstRs['hydra:totalItems'];
-  const maxPage = Math.trunc(totalItems / itemsPerPage) + (totalItems % itemsPerPage === 0 ? 0 : 1);
-
-  return Promise
-    .all(
-      [...Array(maxPage+1).keys()]
-      .slice(2)
-      .map(page => fetch(page))
-    )
-    .then(results => results.reduce((acc, rs) => acc.concat(rs['hydra:member']), firstRs['hydra:member']));
+  const totalItems = firstPageData['hydra:totalItems'];
+  const maxPage =
+    Math.trunc(totalItems / itemsPerPage) +
+    (totalItems % itemsPerPage === 0 ? 0 : 1);
+
+  return Promise.all(
+    [...Array(maxPage + 1).keys()].slice(2).map(page => fetch(page)),
+  )
+    .then(results => {
+      // Check if any page has an error
+      const errorResult = results.find(result => result.error);
+      if (errorResult) {
+        return { error: errorResult.error };
+      }
+
+      // Combine all data from successful results
+      const combinedData = results.reduce(
+        (acc, result) => acc.concat(result.data['hydra:member']),
+        firstPageData['hydra:member'],
+      );
+
+      return { data: combinedData };
+    })
+    .catch(error => {
```

Comment:
Warning, just to take into account: We are changing the behaviour here.

The returned promise it's always resolved/truty because of this last `catch`.
The "outside world" would need to only depend on knowing that an "error" happened because the object `{error: ...}` is returned (instead of a rejected promise or an `await` exception).